extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -25,12 +25,33 @@ export class WorkspaceService implements IWorkspaceService {
 
 	private openedProjects: vscode.Uri[] | undefined = undefined;
 	private excludedProjects: string[] | undefined;
+	private _isProjectProviderAvailable: boolean = false;
 
 	constructor() {
 		Logger.log(`Calling getProjectsInWorkspace() from WorkspaceService constructor`);
+		this.checkIfProjectProviderAvailable();
 		this.getProjectsInWorkspace(undefined, true).catch(err => Logger.error(`Error initializing projects in workspace ${err}`));
 	}
 
+	get isProjectProviderAvailable(): boolean {
+		return this._isProjectProviderAvailable;
+	}
+
+	public checkIfProjectProviderAvailable(): void {
+		Logger.log(`Checking ${vscode.extensions.all.length} extensions to see if there is a project provider is available`);
+		const startTime = new Date().getTime();
+		for (const extension of vscode.extensions.all) {
+			const projectTypes = extension.packageJSON.contributes && extension.packageJSON.contributes.projects as string[];
+			if (projectTypes && projectTypes.length > 0) {
+				Logger.log(`Project provider found. Total time = ${new Date().getTime() - startTime}ms`);
+				this._isProjectProviderAvailable = true;
+			}
+		}
+
+		Logger.log(`No project providers found. Total time = ${new Date().getTime() - startTime}ms`);
```

Comment:
Isn't this going to make the function set this to false? With the change suggested above I'd just make the log message say how long it took and whether it found a provider since the var will be set above.

Reply:
whoops I forgot the return when I copied the old code then swapped out the `returns true` for setting `this._isProjectProviderAvailable = true;`. Thanks for catching it! I'll run another ad hoc build to double check this works too